client/dashboard/sites/staging-site-sync-modal/index.tsx

```diff
@@ -0,0 +1,275 @@
+				description: __(
+					'Pulling will replace the existing files and database of the staging site. An automatic backup of your environment will be created, allowing you to revert changes from the <a>Activity log</a> if needed.'
+				),
+				syncFrom: 'production',
+				syncTo: 'staging',
+			},
+			production: {
+				title: __( 'Pull from Staging' ),
+				description: __(
+					'Pulling will replace the existing files and database of the production site. An automatic backup of your environment will be created, allowing you to revert changes from the <a>Activity log</a> if needed.'
+				),
+				syncFrom: 'staging',
+				syncTo: 'production',
+			},
+			fromLabel: __( 'Pull' ),
+			toLabel: __( 'To' ),
+			learnMore: __( 'Read more about <a>environment pull</a>.' ),
+			submit: __( 'Pull' ),
+		};
+	}
+
+	return {
+		staging: {
+			title: __( 'Push to Production' ),
+			description: __(
+				'Pushing will replace the existing files and database of the production site. An automatic backup of your environment will be created, allowing you to revert changes from the <a>Activity log</a> if needed.'
+			),
+			syncFrom: 'staging',
+			syncTo: 'production',
+		},
+		production: {
+			title: __( 'Push to Staging' ),
+			description: __(
+				'Pushing will replace the existing files and database of the staging site. An automatic backup of your environment will be created, allowing you to revert changes from the <a>Activity log</a> if needed.'
+			),
+			syncFrom: 'production',
+			syncTo: 'staging',
+		},
+		fromLabel: __( 'Push' ),
+		toLabel: __( 'To' ),
+		learnMore: __( 'Read more about <a>environment push</a>.' ),
+		submit: __( 'Push' ),
+	};
+};
+
+export default function StagingSiteSyncModal( {
+	onClose,
+	syncType,
+	environment,
+	productionSiteId,
+	stagingSiteId,
+}: StagingSiteSyncModalProps ) {
+	const syncConfig = getSyncConfig( syncType );
+	const [ domainConfirmation, setDomainConfirmation ] = useState( '' );
+
+	const targetEnvironment = syncConfig[ environment ].syncTo;
+	const sourceEnvironment = syncConfig[ environment ].syncFrom;
+
+	const { data: productionSite } = useQuery( {
+		...siteByIdQuery( productionSiteId ?? 0 ),
+		enabled: !! productionSiteId,
+	} );
+	const productionSiteSlug = productionSite?.slug;
+	const productionSiteTitle = productionSite?.name;
+
+	const { data: stagingSite } = useQuery( {
+		...siteByIdQuery( stagingSiteId ?? 0 ),
+		enabled: !! stagingSiteId,
+	} );
+	const stagingSiteSlug = stagingSite?.slug;
+	const stagingSiteTitle = stagingSite?.name;
+
+	const targetSiteSlug = targetEnvironment === 'production' ? productionSiteSlug : stagingSiteSlug;
+
+	const sourceSiteTitle = sourceEnvironment === 'staging' ? stagingSiteTitle : productionSiteTitle;
+	const targetSiteTitle =
+		targetEnvironment === 'production' ? productionSiteTitle : stagingSiteTitle;
+
+	// TODO: Uncomment once we need it.
+	// const querySiteId = sourceEnvironment === 'staging' ? stagingSiteId : productionSiteId;
```

Comment:
Kudos for adding these statements commented out instead of removing them. I think these will come very handy later when we'll need to connect these actions back. 👍